ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueConsumption.scala

```diff
@@ -261,6 +295,23 @@ object KeyValueConsumption {
     val hexTxId = parseLedgerString("TransactionId")(
       BaseEncoding.base16.encode(entryId.toByteArray)
     )
+
+    val maybeBlindingInfoWithDivulgedContracts =
+      if (txEntry.hasBlindingInfo)
+        Some(Conversions.decodeBlindingInfo(txEntry.getBlindingInfo))
+      else
+        None
+
+    maybeBlindingInfoWithDivulgedContracts.foreach { case (_, divulgedContracts) =>
+      checkDivulgence(divulgedContracts, hexTxId)
```

Comment:
Couldn't you extract the divulged contracts from the blinding info here (through an utility perhaps), rather than having the decoding return two things, one derived from the other one? I think it would be cleaner.

Reply:
Yes, I can. But then the decoding would not be invertible with the encoding. I don't know if it has any impact in the code somewhere if I would allow this asymmetry

Reply:
Oh I see. I don't think it would and I don't see it as an important property in this case but it's really up to you.

Reply:
Ok, split the concerns.